index.js

```diff
@@ -518,7 +518,9 @@
 
   //  Boolean$prototype$equals :: Boolean ~> Boolean -> Boolean
   function Boolean$prototype$equals(other) {
-    return typeof other === typeof this && other.valueOf() === this.valueOf();
+    return typeof this === 'object' ?
+      equals(this.valueOf(), other.valueOf()) :
+      this === other;
   }
```

Comment:
In all three functions the `'object'` branch is simply `equals(this.valueOf(), other.valueOf())`. This consistency pleases me. :)